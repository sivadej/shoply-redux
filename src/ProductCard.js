import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: { minWidth: 275 },
  title: { fontSize: 14 },
  pos: { marginBottom: 12 },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
});

const ProductCard = ({id, product, add, remove}) => {
  const { name, price, description, image_url} = product;
  const classes = useStyles();

  console.log(`rendering ProductCard component for ${name}`);

  return (
    <div>
      <Card xs={12} className={classes.root} variant='outlined'>
        <CardContent>
        <CardHeader
        title={name.toUpperCase()}
        subheader={price}
        />
        <CardMedia
        className={classes.media}
        image={image_url}
        title={name}
      />
          <Typography paragraph>
            {description}
          </Typography>
        </CardContent>
        <CardActions>
          <Button size='small' color='primary' onClick={()=>add(id)}>Add to Cart</Button>
          <Button size='small' color='primary' onClick={()=>remove(id)}>Remove from Cart</Button>
        </CardActions>
      </Card>
    </div>
  )
}

export default ProductCard;