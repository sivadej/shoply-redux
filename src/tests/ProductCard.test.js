import React from 'react';
import { render } from '@testing-library/react';
import ProductCard from './../ProductCard';

const mockProduct = {
  name: 'test item',
  price: 999.99,
  description: 'test item description',
  image_url: 'https://via.placeholder.com/350x300',
};

it('renders without crashing', ()=> {
  render(<ProductCard product={mockProduct} id={12345} />);
});

it('matches snapshot', ()=> {
  const { asFragment } = render(<ProductCard product={mockProduct} id={12345} />);
  expect(asFragment()).toMatchSnapshot();
});

it('capitalizes product name', ()=> {
  const { getByText } = render(<ProductCard product={mockProduct} id={12345} />);
  const productText = getByText("TEST ITEM");
  expect(productText).toBeInTheDocument();
})