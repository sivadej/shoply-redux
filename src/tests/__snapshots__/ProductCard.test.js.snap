// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-118 MuiPaper-outlined MuiPaper-rounded"
      xs="12"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
            >
              TEST ITEM
            </span>
            <span
              class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
            >
              999.99
            </span>
          </div>
        </div>
        <div
          class="MuiCardMedia-root makeStyles-media-121"
          style="background-image: url(https://via.placeholder.com/350x300);"
          title="test item"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
        >
          test item description
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add to Cart
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Remove from Cart
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
